.github/workflows/build-test.yml

```diff
@@ -280,54 +280,78 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        jupyter: [raw, local]
-        python: [nonConda, conda]
-        pythonVersion: ['3.9', '3.10']
-        webOrDesktop: ['']
+        jupyterConnection: [raw] # valid values include raw = uzing zqm, local = using local jupyter, remote = using remote jupyter, web = using remote jupyter in web mode
+        python: [python]  # valid values include python,conda,noPython
+        pythonVersion: ['3.10']
         # Whether we're using stable (<empty>) or pre-release versions of Python packages.
         # When installing pre-release versions, we're only focused on jupyter & related packages.
         # Not pre-release versions of pandas, numpy or other such packages that are not core to Jupyter.
         packageVersion: ['']
+        tags: ['^[^@]+$|@mandatory', '@kernelCore', '@widgets', '@iw', '@webview|@export|@lsp|@variableViewer', '@debugger', '@kernelPicker|@python|@jupyter', '@kernelInstall']
         # We're not running CI on macOS for now because it's one less matrix entry to lower the number of runners used,
         # macOS runners are expensive, and we assume that Ubuntu is enough to cover the UNIX case.
         os: [ubuntu-latest]
         # More details on includes/excludes can be found here https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#example-including-additional-values-into-combinations
         # Basically with exclude, you can exclude any of the combinations from the result matrix.
         # & with include, you can include additional items to the result matrix.
         include:
-          - jupyter: remote # We don't care how remote is started (we only want to test connecting to it)
-            python: nonConda
-            pythonVersion: '3.9'
-            webOrDesktop: 'desktop'
-            os: ubuntu-latest
-          - jupyter: remote
-            python: nonConda
-            pythonVersion: '3.9'
-            webOrDesktop: 'web'
+        # Mandatory tests
+          - jupyterConnection: remote
```

Comment:
@DonJayamanne can you give us a deep dive of how the `include` works with matrix? We now have 17 jobs and it's hard to know how each job is generated by reading the matrix, include and when clauses in the steps down below.

Reply:
> when clauses in the steps down below.

Can you give me an example if thsi

Reply:
I'm referring to https://github.com/microsoft/vscode-jupyter/pull/11867/files#diff-063ca77e012f959eba648db60e6868875fd1e70e5f5ac081193d848f8d61ef32R633 the `if` clauses. It's not easy to know how a job is generated (which makes it hard to troubleshoot locally)

Reply:
Maybe we can also go through the new job list and each owner can check whether the prior steps are required for their tests and investigate how we can simplify each job.

Reply:
Aah yes, unfortunately thats because we need different CLIs to run each test,
E.g. running web tests and running standard tests require different CLIs, see `run: npm run testWebExtension`